sdk/keyvault/azure-security-keyvault-keys/migration_guide.md

```diff
@@ -0,0 +1,226 @@
+
+To try and improve the development experience across Azure services, a set of uniform [design guidelines](https://azure.github.io/azure-sdk/general_introduction.html) was created for all languages to drive a consistent experience with established API patterns for all services. A set of [Java-specific guidelines](https://azure.github.io/azure-sdk/java_introduction.html) was also introduced to ensure that Java clients have a natural and idiomatic feel with respect to the Java ecosystem. Further details are available in the guidelines for those interested.
+
+### Cross Service SDK improvements
+The modern Key Vault Key client library also provides the ability to share in some of the cross-service improvements made to the Azure development experience, such as:
+
+- Using the new Azure Identity library to share a single authentication approach between clients.
+- A unified logging and diagnostics pipeline offering a common view of the activities across each of the client libraries.
+
+## Important changes
+### Separate packages and clients
+In the interest of simplifying the API, `azure-keyvault` and `KeyVaultClient` were split into separate packages and clients:
+
+- [`azure-security-keyvault-certificates`](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/keyvault/azure-security-keyvault-certificates/README.md) contains `CertificateClient` for working with certificates.
+- `azure-security-keyvault-keys` contains `KeyClient` for working with keys and `CryptographyClient` for performing cryptographic operations.
+- [`azure-security-keyvault-secrets`](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/keyvault/azure-security-keyvault-secrets/README.md) contains `SecretClient` for working with secrets.
+
+### Package names and namespaces
+Package names and the namespace root for the modern Azure client libraries for Java have changed. Each will follow the pattern `com.azure.<area>.<service>` where the legacy clients followed the pattern `com.microsoft.azure.<service>`. This provides a quick and accessible means to help understand, at a glance, whether you are using the modern or legacy clients.
+
+In the case of the Key Vault, the modern client libraries have packages and namespaces that begin with `com.azure.security.keyvault` and were released beginning with version `4.0.0`. The legacy client libraries have packages and namespaces that begin with `com.microsoft.azure.keyvault` and a version of `1.x.x` or below.
+
+### Client constructors
+Across all modern Azure client libraries, clients consistently take an endpoint or connection string along with token credentials. This differs from `KeyVaultClient`, which took an authentication delegate and could be used for multiple Key Vault endpoints.
```

Comment:
Shouldnt we also talk about constructors vs builders here?

Reply:
You are right, this is a big change we made for all of Track 2 in Java. I will make sure to add it.